### ISSUE TITLE
Automatically add static release notes in generate_appcast

### ISSUE THREAD

**Issue by zorgiepoo** [sparkle-project/Sparkle#2661]
### Discussed in https://github.com/sparkle-project/Sparkle/discussions/2660

<div type='discussions-op-text'>

<sup>Originally posted by **vojtabohm** November 11, 2024</sup>
Hey,

I am curious on how to implement release notes so it's as pleasant as possible. I started publishing the release notes at https://steerapp.ai/changelog and I want to embed those on every update. I don't want to tailor custom html files etc. based on a version. Just always load the same change log (which I will update on my website).

However I cannot do it in the `generate_appcast` script. It allows `--release-notes-url-prefix` where I put the URL above but it doesn't add it if you don't have a custom html file for the update as well. Which I do not want. I want each update to have the same releaseNotes link. Right now I manually add it in the xml for each version which is a bit tedious and also, generate_appcast will delete it if I run it again (when creating a new update) which is frustrating.

Any help is appreciated.

Also: fullReleaseNotes is not what is show to the user when the 'a new update is available' window is shown.</div>

**Comment by zorgiepoo** [comment]
This sounds like a reasonable change to generate_appcast to me.

I don't think generate_appcast should update the release notes URL of existing items which already have one present (this was done when `release-notes-url-prefix` was added to update the URLs of existing items, but I think this was the wrong tradeoff).
It makes sense to have an option to use a static URL as well for newly inserted updates.

As of now if you use generate_appcast I can't think of better workarounds than what you're already discovered, probably. Potentially you could have separate release note files but they could actually be symlinks or copies to the one file you care about on the server or something.

**Comment by vojtabohm** [comment]
Yeah, I ended up creating a simple python file to add them to all updates.

**Comment by zorgiepoo** [comment]
Now that I added support for signing appcasts and release notes (#2822) this moves towards more of the direction of always updating release notes for each item, which I guess also makes sense if the url-prefix changes, rather than only inserting a release notes for each new inserted update item.

In practice, release note files can change even if the same one is used for a different range of versions. E.g. release note file for version 1.x may differ from 2.x, which has different minimum system version requirements.

Anyway, I don't think this issue is going to be easily addressed. I think making copies or using symlinks per item is fine.
